public/app/features/dashboard/components/AnnotationSettings/AnnotationSettingsEdit.tsx

```diff
@@ -73,8 +73,15 @@ export const AnnotationSettingsEdit = ({ editIdx, dashboard }: Props) => {
 
   const onDataSourceChange = (ds: DataSourceInstanceSettings) => {
     onUpdate({
-      ...annotation,
```

Comment:
what if we change to a different data source of the same type?

Reply:
good point, pushing in a sec